### PR TITLE
feat: implement improved rollup backend

### PR DESCRIPTION
**Pull request by atarashansky**
## Reason for Change

- #6400
- Implementing a new rollup algorithm as part of the performance improvement epic.

## Changes

- Replaced the existing rollup algorithm with a simpler version that leverages pandas explode and groupby to perform the rollup.
- Make some QOL improvements to the performance profiling workflow

## Testing steps

- Unit tests updated
- Rdev inspected to confirm equivalence of data in the frontend.
- performance profiling was done on this branch vs #6427 (which is a copy of `main`). 
[scatter_plots_20240108_132010.pdf](https://github.com/chanzuckerberg/single-cell-data-portal/files/13866359/scatter_plots_20240108_132010.pdf)

### CODE REVIEW

File: backend/wmg/api/common/rollup.py
Comment (atarashansky) [inline backend/wmg/api/common/rollup.py:1]: Core changes are here.

File: backend/wmg/api/v2.py [hunk 1]
```diff
@@ -101,9 +101,25 @@ def query():
...
+                # from the gene expression dataframe.
+                rolled_gene_expression_df = rollup(
+                    gene_expression_df, snapshot.cell_type_ancestors, filter_redundant_nodes=False
                 )
+                rolled_cell_counts_grouped_df = rollup(cell_counts_grouped_df, snapshot.cell_type_ancestors)
+
+                # filter out rows that do not have a corresponding cell type in the cell counts dataframe
+                # as these will not be displayed in the UI anyway
+
+                with ServerTiming.time("filter out rows"):
```
Comment (atarashansky) [inline backend/wmg/api/v2.py:115]: This block filters out redundant cell types from gene expression by checking to see what's not included in the cell counts dataframe, since the cell counts rollup DOES filter out redundant cell types (see above comment for more info).

File: backend/wmg/pipeline/cell_type_ancestors.py [hunk 2]
```diff
@@ -30,8 +30,7 @@ def create_cell_type_ancestors(corpus_path: str) -> None:
     valid_cell_types = cell_counts_df["cell_type_ontology_term_id"].unique()
     ancestors_dict = {}
     for cell_type_id in valid_cell_types:
-        cell_type_ancestors = sorted(set(ancestors(cell_type_id)).intersection(valid_cell_types))
-        ancestors_dict[cell_type_id] = ",".join(cell_type_ancestors)
+        ancestors_dict[cell_type_id] = sorted(set(ancestors(cell_type_id)).intersection(valid_cell_types))
```
Comment (atarashansky) [inline backend/wmg/pipeline/cell_type_ancestors.py:33]: the pipeline changes that were previously merged in incorrectly stored the ancestors as a comma-separated string, but we need it as a python list.

File: frontend/tests/features/wheresMyGene/wheresMyGene.test.ts [hunk 3]
```diff
@@ -451,7 +451,7 @@ describe("Where's My Gene", () => {
 
       // Click into a cell type that has marker genes
       await page
-        .getByTestId("cell-type-info-button-adipose tissue-phagocyte")
+        .getByTestId("cell-type-info-button-adipose tissue-contractile cell")
```
Comment (atarashansky) [inline frontend/tests/features/wheresMyGene/wheresMyGene.test.ts:454]: phagocytes are no longer in adipose tissue in the latest version of the snapshot. switching to contractile cell.

this is not related to any of the rollup changes since none of the existing pipeline code was changed. a new pipeline step was simply added.

File: requirements-backend.txt [hunk 4]
```diff
@@ -44,5 +44,5 @@ setproctitle==1.3.2 # for gunicorn integration with datadog
 SQLAlchemy-Utils>=0.36.8
 SQLAlchemy>=1.4.0,<1.5
 tenacity
-tiledb==0.22.3  # Portal's tiledb version should always be the same or older than Explorer's
+tiledb==0.24.0
```
Comment (atarashansky) [inline requirements-backend.txt:47]: update backend version to match pipeline

File: scripts/compare_wmg_profiling_results.py
Comment (atarashansky) [inline scripts/compare_wmg_profiling_results.py:1]: This is now included as a feature in the performance profiling script

**Review by prathapsridharan — APPROVED** [review]
LGTM
